scripts/iam/README.md

```diff
@@ -0,0 +1,202 @@
+# GenAI IDP IAM Roles
+
+This directory contains IAM role definitions for operating GenAI IDP solution patterns with least-privilege access.
+
+## All Patterns Deployer Role
+
+The `all-patterns-deployer-role.yaml` CloudFormation template creates a least-privilege IAM role for deploying all three GenAI IDP patterns (1, 2, and 3). This role is designed for users who need to deploy and manage patterns but don't have administrator rights.
+
+### Key Features:
+- **Tag-based access control** - Users must be in the deployer group and use required tags
+- **All patterns support** - Can deploy Pattern 1 (BDA), Pattern 2 (Textract+Bedrock), and Pattern 3 (UDOP)
+- **CloudFormation compatible** - Can be used with `--role-arn` parameter
+- **Least privilege** - Only permissions needed for pattern deployment
+- **Secure by design** - Multiple layers of access control (group membership + session tags)
+
+### Deployment:
+```bash
+aws cloudformation create-stack \
+  --stack-name genai-idp-deployer-role \
+  --template-body file://iam/all-patterns-deployer-role.yaml \
+  --parameters ParameterKey=MasterStackName,ParameterValue=YOUR_MASTER_STACK_NAME \
+               ParameterKey=DeployerGroupName,ParameterValue=GenAI-IDP-Deployers \
+  --capabilities CAPABILITY_NAMED_IAM
+```
+
+### Usage:
+```bash
+# Add users to the group
+aws iam add-user-to-group --group-name GenAI-IDP-Deployers --user-name username
```

Comment:
username should be USERNAME to be consistent with syntax indicating that this should be modified